samples/musicfestival-frontend-react/src/components/SearchButton.tsx

```diff
@@ -1,78 +1,113 @@
     const [isShown, setIsShown] = useState(false)
     const [searchValue, setSearchValue] = useState<CustomString>(searchParams.get("q")?.toString() ?? "")
+    const [ranking, setRanking] = useState<Ranking>(Ranking[capitalize(searchParams.get("r")?.toString().toLowerCase() || "") as keyof typeof Ranking] || Ranking.Relevance)
     const [orderBy] = useState("ASC")
-    let autocompleteData : ArtistAutocompleteQuery | undefined = undefined
-    
+    let autocompleteData: ArtistAutocompleteQuery | undefined = undefined
+
     let modeEdit = isEditOrPreviewMode()
-    let variables = generateGQLSearchQueryVars(token, window.location.pathname, searchValue as string | null, orderBy);
-    const { data : artistAutocompleteData } = useArtistAutocompleteQuery({ endpoint: singleKeyUrl }, variables, { staleTime: 2000, enabled: !modeEdit || !!token })
+    let variables = generateGQLSearchQueryVars(token, window.location.pathname, searchValue as string | null, orderBy, ranking);
+    const {data: artistAutocompleteData} = useArtistAutocompleteQuery({endpoint: singleKeyUrl}, variables, {
+        staleTime: 2000,
+        enabled: !modeEdit || !!token
+    })
     autocompleteData = artistAutocompleteData
-    
-    function search(event: any, action: string){
-        if ((action == "keypress" && event.charCode === 13) || action == "buttonclick") {
-            window.location.href = `${window.location.origin}/search?q=${searchValue}&f=${filterValue ?? ARTIST}`
+
+    const onSearch = (event: any) => {
+        window.location.href = `${window.location.origin}/search?q=${searchValue}&f=${filterValue ?? ARTIST}&r=${ranking}`;
+    }
+
+    const onPressSearch = (event: any) => {
+        if (event.key === 'Enter' || event.keyCode === 13) {
+            onSearch(event);
         }
     }
 
-    function onValueChange(event: any){
+    const onValueChange = (event: any) => {
         setSearchValue(event.target.value);
         event.target.value !== undefined ? setIsShown(true) : setIsShown(false);
     }
 
-    function onAutoClick(event: any){
+    const onAutoClick = (event: any) => {
         setSearchValue(event.target.textContent);
-        window.location.href = `${window.location.origin}/search?q=${event.target.textContent}&f=${filterValue ?? ARTIST}`
+        window.location.href = `${window.location.origin}/search?q=${event.target.textContent}&f=${filterValue ?? ARTIST}&r=${ranking}`;
+    }
+
+    const onChangeSearchType = (event: any) => {
+        if (event.target.checked) {
+            setRanking(Ranking.Semantic);
+        } else {
+            setRanking(Ranking.Relevance);
+        }
     }
 
+    useEffect(() => {
+        if (isFirstLoad.current) {
+            isFirstLoad.current = false;
+        } else {
+            if (searchValue) {
+                window.location.href = `${window.location.origin}/search?q=${searchValue}&f=${filterValue ?? ARTIST}&r=${ranking}`;
+            }
+        }
+    }, [ranking]);
```

Comment:
Effect is executed at the start when the component is rendered, so we can consider moving this to the top of the component code. Just a minor thing.